### DISCUSSION TITLE
Unable to pull down Git repository

### DISCUSSION THREAD

**Discussion by aiqubits** [TabbyML/tabby#2942] [Q&A]
**Describe the bug**
Context Providers Unable to pull down Git repository.
![image](https://github.com/user-attachments/assets/00d6771a-3cdf-409c-9cbc-d8dd52816df4)
![image](https://github.com/user-attachments/assets/291c1b39-3664-49e5-a267-9032fabc3ded)
![image](https://github.com/user-attachments/assets/e61acfe6-b7ab-48ba-9e36-b895ae09dc76)
![image](https://github.com/user-attachments/assets/b2cf33be-1af1-4eeb-ba89-f6171dbcbad2)

**Information about your version**
$ tabby --version
tabby 0.14.0
![image](https://github.com/user-attachments/assets/92727afe-8c25-49ce-88c3-e7227bafef2a)

**Information about your GPU**
GeForce RTX 3060
![image](https://github.com/user-attachments/assets/8f3cd3c1-fdde-4f57-b2cc-1fa4f9a681ad)

**Additional context**
Every time a job is executed, sometimes it succeeds, but most of it fails and cannot be pulled down from the Git repository.

**Comment by wsxiaoys** [comment]
Hi - could you check your network configuration to ensure you have a good connection?

I've added rust repo to https://demo.tabbyml.com/files/git/rust/-/tree/master and it's working fine.

**Comment by aiqubits** [comment]
> Hi - could you check your network configuration to ensure you have a good connection?
> 
> I've added rust repo to https://demo.tabbyml.com/files/git/rust/-/tree/master and it's working fine.

The problem I encountered is that some repos can be pulled down while others cannot, and every time Tabby is restarted, the existing repos will be cleared. How can I set up the dropdown repos to only be updated and not cleared? Another question is, can I manually execute "git pull repo" and change the directory name of the repo so that Tabby can recognize it by itself? Will it work？

**Comment by wsxiaoys** [comment]
In that case, could you provide more logs generated by the Tabby process, especially those related to the error encountered during cloning/pulling the repository?

**Comment by aiqubits** [comment]
> In that case, could you provide more logs generated by the Tabby process, especially those related to the error encountered during cloning/pulling the repository?

![image](https://github.com/user-attachments/assets/d4c8cd05-f834-47b7-83f9-c0d9038c583f)

**Comment by wsxiaoys** [comment]
Based on the error log, it is very likely that the problem is simply caused by your network environment.
